### PR TITLE
Update and improve Italian translation

### PR DESCRIPTION
**Pull request by zenseii**
Add latest strings and make much needed improvements.

Many lines changed due to recent artifact code overhaul.

### CODE REVIEW

File: files/lang/it.po [hunk 1]
```diff
@@ -5937,10 +6073,10 @@ msgid "Eye of the Magi"
 msgstr "Occhio del Mago"
 
 msgid "Mermaid"
-msgstr "Sirenetta"
+msgstr "Sirena"
 
 msgid "Sirens"
-msgstr "Sirene"
+msgstr "Arpie"
```
Comment (zenseii) [inline files/lang/it.po:6079]:
```suggestion
msgstr "Sirene"
```

File: files/lang/it.po [hunk 2]
```diff
@@ -5937,10 +6073,10 @@ msgid "Eye of the Magi"
 msgstr "Occhio del Mago"
 
 msgid "Mermaid"
-msgstr "Sirenetta"
+msgstr "Sirena"
```
Comment (zenseii) [inline files/lang/it.po:6076]:
```suggestion
msgstr "Sirenetta"
```

File: files/lang/it.po [hunk 3]
```diff
@@ -7892,9 +8452,3 @@ msgstr ""
 "Scopri l'ultima versione del gioco su\n"
 "https://github.com/ihhub/\n"
 "fheroes2/releases"
-
-#~ msgid "Buy Monsters"
```
Comment (ihhub) [inline files/lang/it.po:7896]: Can we please remove these commented lines since they aren't in use?

Comment (zenseii) [reply files/lang/it.po:7896]: If you mean the lines like :
```#~ msgid "Buy Monsters"```
Then they are already removed and are colored red.

Comment (ihhub) [reply files/lang/it.po:7896]: Ah, my bad.

**Comment by ihhub** [comment]
@zenseii , thank you so much for your huge efforts towards translation support!

**Pull request merged**